src/tools/ScenarioMeasurement/Startup/Startup.cs

```diff
@@ -0,0 +1,178 @@
+﻿using Microsoft.Diagnostics.Tracing.Parsers;
+using Microsoft.Diagnostics.Tracing.Session;
+using Reporting;
+using System;
+using System.IO;
+using System.Linq;
+
+namespace ScenarioMeasurement
+{
+    enum MetricType
+    {
+        TimeToMain,
+        GenericStartup,
+        WPF
+    }
+    class Startup
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="appExe">Full path to test executable</param>
+        /// <param name="metricType">Type of interval measurement</param>
+        /// <param name="scenarioName">Scenario name for reporting</param>
+        /// <param name="processWillExit">true: process exits on its own. False: process does not exit, send close.</param>
+        /// <param name="iterations">Number of measured iterations</param>
+        /// <param name="timeout">Max wait for process to exit</param>
+        /// <param name="measurementDelay">Allowed time for startup window</param>
+        /// <param name="appArgs">optional arguments to test executable</param>
+        /// <param name="logFileName">optional log file. Default is appExe.startup.log</param>
+        /// <param name="workingDir">optional working directory</param>
+        /// <param name="warmup">enables/disables warmup iteration</param>
+        /// <param name="traceFileName">trace file name</param>
+        /// <param name="guiApp">true: app under test is a GUI app, false: console</param>
+        /// <param name="skipProfileIteration">true: skip full results iteration</param>
+        /// <param name="reportJsonPath">path to save report json</param>
+        /// <returns></returns>
+        static int Main(string appExe,
+                        MetricType metricType,
+                        string scenarioName,
+                        string traceFileName,
+                        bool processWillExit = false,
+                        int iterations = 5,
+                        int timeout = 60,
+                        int measurementDelay = 15,
+                        string appArgs = "",
+                        string logFileName = "",
+                        string workingDir = "",
+                        bool warmup = true,
+                        bool guiApp = true,
+                        bool skipProfileIteration = false,
+                        string reportJsonPath = "")
+        {
+            Logger logger = new Logger(String.IsNullOrEmpty(logFileName) ? $"{appExe}.startup.log" : logFileName);
```

Comment:
I'm thinking we might also want to have an option to abort all sessions - like ```perfview abort```.  Might also be good to run this at the beginning of any ```startup.exe``` run.

Reply:
Going on the backlog.